pilot/pkg/security/authz/builder/builder.go

```diff
@@ -96,64 +96,54 @@ func New(trustDomainBundle trustdomain.Bundle, push *model.PushContext, policies
 
 // BuildHTTP returns the HTTP filters built from the authorization policy.
 func (b Builder) BuildHTTP() []*hcm.HttpFilter {
-	b.logger = &AuthzLogger{}
-	defer b.logger.Report()
-	if b.option.IsCustomBuilder {
-		// Use the DENY action so that a HTTP rule is properly handled when generating for TCP filter chain.
-		if configs := b.build(b.customPolicies, rbacpb.RBAC_DENY, false); configs != nil {
-			b.logger.AppendDebugf("built %d HTTP filters for CUSTOM action", len(configs.http))
-			return configs.http
-		}
-		return nil
-	}
-
-	var filters []*hcm.HttpFilter
-	if configs := b.build(b.auditPolicies, rbacpb.RBAC_LOG, false); configs != nil {
-		b.logger.AppendDebugf("built %d HTTP filters for AUDIT action", len(configs.http))
-		filters = append(filters, configs.http...)
-	}
-	if configs := b.build(b.denyPolicies, rbacpb.RBAC_DENY, false); configs != nil {
-		b.logger.AppendDebugf("built %d HTTP filters for DENY action", len(configs.http))
-		filters = append(filters, configs.http...)
-	}
-	if configs := b.build(b.allowPolicies, rbacpb.RBAC_ALLOW, false); configs != nil {
-		b.logger.AppendDebugf("built %d HTTP filters for ALLOW action", len(configs.http))
-		filters = append(filters, configs.http...)
-	}
-	return filters
+	return build(b, b.buildHTTP, "HTTP", false)
 }
 
 // BuildTCP returns the TCP filters built from the authorization policy.
 func (b Builder) BuildTCP() []*listener.Filter {
+	return build(b, b.buildTCP, "TCP", true)
+}
+
+// BuildTCPRulesAsHTTPFilter returns the TCP filters built from the authorization policy.
+func (b Builder) BuildTCPRulesAsHTTPFilter() []*hcm.HttpFilter {
+	return build(b, b.buildHTTP, "HTTP for TCP", true)
+}
```

Comment:
It seems like this doesn't do exactly what (I thought) we wanted. It will build HTTP rules normally, but log that they are for TCP I think. Do we want it to perform the same types of L7 attribute stripping that occurs when interpreting workload-attached rules for ztunnel in order to more precisely replicate the sandwiched waypoint behavior?

Reply:
My intent is it builds L4 rules (stripping L7 semantics) but wraps it in an envoy HTTPFilter instead of a Network filter